remove componentwise comparison of complex values in torch.testing.assert_close

pmeier opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* __->__ #63841
* #63572

Closes #61906.

cc @ezyang @gchanan

Differential Revision: [D30633526](https://our.internmc.facebook.com/intern/diff/D30633526)

pmeier committed:
remove componentwise comparison of complex values in torch.testing.assert_close

[ghstack-poisoned]

pmeier committed:
Update on "remove componentwise comparison of complex values in torch.testing.assert_close"


Closes #61906.

cc @ezyang @gchanan

[ghstack-poisoned]

mruberry commented on test/test_testing.py:
```diff
@@ -1180,30 +1194,6 @@ def test_mapping_mismatching_values_msg(self):
-        for fn in assert_close_with_inputs(actual, expected):
-            fn(equal_nan="relaxed")
-
-    def test_matching_conjugate_bit(self):
```
Don't we still need this test?

pmeier replied:
#56058 removed the parts that this test was designed for. So yes, we can keep it, but I don't think we need to. cc @anjali411

mruberry replied:
How about we keep it for now (just in case) but add a TODO: over it that we should review later?

mruberry commented on torch/testing/_asserts.py:
```diff
@@ -429,11 +383,18 @@ def append_difference(msg: str, *, type: str, difference: float, index: Tuple[in
 
     return msg.strip()
 
+def _get_XXX_dtype(dtype: torch.dtype) -> torch.dtype:
```
_get_comparison_dtype?

mruberry replied:
Add a comment for what this function does and how it's intended to be used

pmeier replied:
Oops :innocent:

mruberry reviewed (commented):
Cool simplification but maybe a little too aggressive with the conjugate bit test? I also have a name suggestion (naming things is hard but XXX seems easy to improve on)

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30633526).

pmeier committed:
Update on "remove componentwise comparison of complex values in torch.testing.assert_close"


Closes #61906.

cc @ezyang @gchanan

Differential Revision: [D30633526](https://our.internmc.facebook.com/intern/diff/D30633526)

[ghstack-poisoned]

mruberry reviewed (approved):
Cool!

mruberry commented:
@mruberry has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D30633526).
